### PR TITLE
ISPN-13735 Actions to keep docs in sync

### PR DESCRIPTION
**Pull request by oraNod**
GitHub action to sync changes to documentation to the dev docs.

https://issues.redhat.com/browse/ISPN-13735

### CODE REVIEW

File: .github/workflows/stable_docs.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        cache: maven
+    - name: Build docs with Maven
+      run: mvn install -Pdistribution -pl documentation -am -s maven-settings.xml
+    - name: Copy docs to the site
+      run: |
+        git clone -b $target_branch $target
+        cp -r infinispan/documentation/target/generated/1*/html/* infinispan.github.io/docs/stable/
```
Comment (ryanemerson) [inline .github/workflows/stable_docs.yml:32]: It seems like the only difference between the two GH actions is the `infinispan.github.io` url and the commit message. We can just use a single action for both branches and just update the url depending on the branch.

https://stackoverflow.com/questions/58139406/only-run-job-on-specific-branch-with-github-actions

File: .github/workflows/sync_docs.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    - name: Copy docs to stable
+      run: |
+        git clone -b $target_branch $target
+        cp -r infinispan/documentation/target/generated/1*/html/* infinispan.github.io/docs/stable/
+        cd infinispan.github.io
+        git config --global user.email "dnaro@redhat.com"
+        git config --global user.name "dnaro"
+        git add . --all
+        git commit -m "Synchronized docs to stable"
+        git push origin master
```
Comment (ryanemerson) [inline .github/workflows/sync_docs.yml:61]: I think we can reduce repetition further by doing something like:

```yaml
name: Update dev docs

on:
  push:
    branches:
      - main
      - 13.0.x
    paths:
      - documentation

env:
  target: git@github.com:infinispan/infinispan.github.io.git
  target_branch: master

jobs:
  upload:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
    
    - name: Set up JDK 11
      uses: actions/setup-java@v2
      with:
        java-version: '11'
        distribution: 'temurin'
        cache: maven
    
    - name: Build docs with Maven
      run: mvn install -Pdistribution -pl documentation -am -s maven-settings.xml

    - name: Clone infinispan.github.io
      run: |
        git clone -b $target_branch $target

    - if: github.ref == 'refs/heads/main'
      name: Copy docs to dev
      run: |
        cp -r infinispan/documentation/target/generated/1*/html/* infinispan.github.io/docs/dev/

    - if: github.ref == 'refs/heads/13.0.x'
      name: Copy docs to stable
      run: |
        cp -r documentation/target/generated/1*/html/* infinispan.github.io/docs/stable/
        
    - name: Commit and push changes
      run: |
        cd infinispan.github.io
        git config --global user.email "dnaro@redhat.com"
        git config --global user.name "dnaro"
        git add . --all
        git commit -m "Synchronized docs from infinispan ${{ github.ref }}"
        git push origin master
```

Comment (oraNod) [reply .github/workflows/sync_docs.yml:53]: Oh that is top class. Updated!!

**Comment by ryanemerson** [comment]
Thanks @oraNod

**Pull request merged**